### DISCUSSION TITLE
I can't use pinia outside of component setup

### DISCUSSION THREAD

**Discussion by rangeldor** [vuejs/pinia#664] [Help and Questions] [answered]
When I try to use it in the api.ts I get this error: 
TypeError: Object(...) is not a function

api.ts
![image](https://user-images.githubusercontent.com/51749346/132924662-0b9846fd-7d31-409d-8f1c-c9bafd50ca5a.png)

When I try to use it in the router.ts I get this error:  
Uncaught Error: [🍍]: getActivePinia was called with no active Pinia. Did you forget to install pinia?

const pinia = createPinia()
app.use(pinia)

router.ts
![image](https://user-images.githubusercontent.com/51749346/132924820-6aec68b7-93b9-429b-baec-14a38de86df7.png)

**Comment by BenShelton** [comment, marked as answer]
That error is happening because you are trying to create a store before calling `app.use(pinia)` (or however you are initializing it)

Have a look at the docs here: https://pinia.esm.dev/core-concepts/outside-component-usage.html
Most importantly:
> The easiest way to ensure this is always applied is to defer calls of useStore() by placing them inside functions that will always run after pinia is installed.

I can't see in your screenshots above where exactly you planned on using the userStore, but if it's in an Axios interceptor then call it within the interceptor function:

```ts
axios.interceptors.request.use(function (config) {
  const userStore = useUserStore()
  config.headers.Authorization = userStore.token
  return config
});
```

Or if it's in something like a router guard then you need to call the user store within the guard itself:

```ts
router.beforeEach((to) => {
  const userStore = useUserStore()
  if (!userStore.isLoggedIn) return '/login'
})
```

**Reply by BenShelton** [reply]
You can see a complete example here: https://codesandbox.io/s/pinia-vue-3-axios-interceptor-6jf11

Does this solve your problem @rangeldor ?

**Reply by rangeldor** [reply]
Hello @BenShelton! Thank you for reply.

I have already done it.

The problem is that I don't want put it in everywhere. 

Ex: 
![image](https://user-images.githubusercontent.com/51749346/134811591-d394fab6-9355-49b1-bd20-6e804a787417.png)

If I use it out of the "if", I get error.

**Reply by rangeldor** [reply]
@BenShelton , I used of this form and worked!

![image](https://user-images.githubusercontent.com/51749346/134812978-830da775-b1d1-41cf-8153-dbb622153c6f.png)

Thank you!

**Reply by BenShelton** [reply]
@rangeldor Glad this worked for you.
To explain, you must retrieve the store _within_ the interceptor's callback function but it can be the first thing you do in the callback (as you show above) if you want to access the store within the entire scope of the callback.

**Comment by JoeABCDEF** [comment]
请问 你是使用的 vscode吗  你的主题是啥

**Reply by rangeldor** [reply]
Hello! 

I am using a Synthwave 84 blues from VSCode. 

I enabled neon.

I hope I could have helped :)
